### PR TITLE
XCode build integration

### PR DESCRIPTION
**Pull request by vvvar**
- Added script for building XCode project;
- Fix CMake to include boost_ext into working tree;

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,28 +11,40 @@ SET (BOOST_OPTIONAL_EXT ${CMAKE_CURRENT_SOURCE_DIR})
 include(${CMAKE_CURRENT_SOURCE_DIR}/Build/conanbuildinfo.cmake)
 conan_basic_setup(TARGETS)
 
+# Boost optional extension
+SET (EXT_SRC
+        boost/optional_ext.hpp
+)
+add_library(boost_optional_ext_src ${EXT_SRC})
+set_target_properties(boost_optional_ext_src PROPERTIES LINKER_LANGUAGE CXX)
```
Comment (vvvar) [inline CMakeLists.txt:19]: Need to be used because CMake won't detect linker language based on .hpp file.

File: CMakeLists.txt [hunk 2]
```diff
@@ -11,28 +11,40 @@ SET (BOOST_OPTIONAL_EXT ${CMAKE_CURRENT_SOURCE_DIR})
 include(${CMAKE_CURRENT_SOURCE_DIR}/Build/conanbuildinfo.cmake)
 conan_basic_setup(TARGETS)
 
+# Boost optional extension
+SET (EXT_SRC
+        boost/optional_ext.hpp
+)
+add_library(boost_optional_ext_src ${EXT_SRC})
```
Comment (vvvar) [inline CMakeLists.txt:18]: Added to be visible in the xcode working tree.

File: CMakeLists.txt [hunk 3]
```diff
@@ -11,28 +11,40 @@ SET (BOOST_OPTIONAL_EXT ${CMAKE_CURRENT_SOURCE_DIR})
...
 add_executable(boost_optional_ext_example ${EXAMPLE_SRC})
 target_link_libraries(boost_optional_ext_example CONAN_PKG::boost)
 target_include_directories(boost_optional_ext_example
     PRIVATE
     ${BOOST_INCLUDE_DIRS}
     ${BOOST_OPTIONAL_EXT})
+
+# Group all files under "src" name    
+source_group("src"
+    FILES ${EXT_SRC} ${TEST_SRC} ${EXAMPLE_SRC}
```
Comment (vvvar) [inline CMakeLists.txt:49]: At this point, each executable will have its own "src" group int the project's working tree.

File: CMakeLists.txt [hunk 4]
```diff
@@ -11,28 +11,40 @@ SET (BOOST_OPTIONAL_EXT ${CMAKE_CURRENT_SOURCE_DIR})
...
         tests/test_optional_ext.cpp
         tests/test_optional_ext_with_const.cpp
         tests/test_hof.cpp
         tests/tests_main.cpp
 )
-
-add_executable(boost_optional_ext ${TEST_SRC})
-target_link_libraries(boost_optional_ext CONAN_PKG::boost)
-target_include_directories(boost_optional_ext
+add_executable(boost_optional_ext_test ${TEST_SRC})
```
Comment (vvvar) [inline CMakeLists.txt:28]: Renamed to have a clear semantic separation between targets.

**Review by vgordievskiy — APPROVED** [review]

**Pull request merged**